sdk/src/akkaserverless.ts

```diff
@@ -0,0 +1,454 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import * as fs from 'fs';
+import * as path from 'path';
+import * as grpc from '@grpc/grpc-js';
+import * as settings from '../settings';
+
+import * as discovery from '../proto/akkaserverless/protocol/discovery_pb';
+import * as discovery_grpc from '../proto/akkaserverless/protocol/discovery_grpc_pb';
+import * as google_protobuf_empty_pb from 'google-protobuf/google/protobuf/empty_pb';
+import { PackageInfo } from './package-info';
+
+export class Bindings {
+  readonly address: string;
+  readonly port: number;
+
+  constructor(
+    address?: string | null | undefined,
+    port?: number | null | undefined,
+  ) {
+    const portNumber = () => {
+      if (process.env.PORT) {
+        return parseInt(process.env.PORT);
+      } else {
+        return undefined;
+      }
+    };
+    this.address = address || process.env.HOST || '127.0.0.1';
+    this.port = port || portNumber() || 8080;
+  }
+}
+
+function loadJson(filename: string) {
+  return JSON.parse(fs.readFileSync(filename).toString());
+}
+
+const userPkgJson = path.join(process.cwd(), 'package.json');
+
+class ServiceInfo {
+  readonly name: string;
+  readonly version: string;
+  private pkgName: string = 'unknown';
+  private pkgVersion: string = '0.0.0';
+
+  constructor(name?: string, version?: string, filename: string = userPkgJson) {
+    if (!name || !version) {
+      this.loadFromPkg(filename);
+    }
+    this.name = name || this.pkgName;
+    this.version = version || this.pkgVersion;
+  }
+
+  private loadFromPkg(filename: string = userPkgJson) {
+    const json = loadJson(filename);
+    this.pkgName = json.name;
+    this.pkgVersion = json.version;
+  }
+}
+
+export interface Service {
+  componentType: () => string;
+  register: (server: grpc.Server) => void;
+}
+
+export interface EntityPassivationStrategy {
+  timeout?: number;
+}
+
+export interface ComponentOptions {
```

Comment:
In this case, we can't directly rely on the Protocol types, since additional fields are included ( e.g. `includeDirs`). Not judging the current design, just making this PR as less breaking as possible...

Reply:
Ok, I think I misinterpreted this to be about the duck-typey objects in the old discovery logic where it was internal but it's rather is a way to pass the options from components to the discovery?